packages/flutter/lib/src/semantics/semantics_event.dart

```diff
@@ -121,3 +124,33 @@ class ScrollCompletedSemanticsEvent extends SemanticsEvent {
     return map;
   }
 }
+
+/// An event for a semantic announcement.
+///
+/// This should be used for announcement that are not seamlessly announced by
+/// the system as a result of a UI state change.
+///
+/// For example a camera application can use this method to make accessibility
+/// announcements regarding objects in the viewfinder.
+class AnnounceSemanticsEvent extends SemanticsEvent {
+
+  /// Constructs an event that triggers an announcement by the platform.
+  const AnnounceSemanticsEvent(this.message, this.textDirection) :
+    assert(message != null),
+    assert(textDirection != null),
+    super('announce');
+
+  /// The message to announce. cannot be null.
+  final String message;
+
+  /// Text direction for [message]. cannot be null.
+  final TextDirection textDirection;
+
```

Comment:
for both of those, consider if maybe the "Cannot be null" (maybe better as "This property must not be null") should be on a separate paragraph, so that it isn't visible in the api docs overview for the class.